Added RapidXML package

ErniBrown opened the pull request:
I created the RapidXML package, as proposed by @roalz in #546

ruslo commented:
This library is header-only as far as I understand. This will be a trivial task to add CMake configs for it. It's much better than introducing new scheme and keeping the package unrelocatable. Few forks mentioned in https://github.com/ruslo/hunter/issues/546 @ErniBrown have you looked at them?

ErniBrown commented:
I'm not sure I understand, should we use a cmake project with an integrated config?
In #546 a cmake project is proposed, but no config is available with it.
What's your suggestion about this package?
Thanks

ruslo commented:
> I'm not sure I understand, should we use a cmake project with an integrated config?

We should patch the project and add CMake build. Since it's a header-only library that will be pretty simple step. Here is the boilerplate code to copy:
* https://github.com/ruslo/leathers/blob/1cf363e8abb440973cb400f421da8a3706a6b030/CMakeLists.txt#L91-L130

> In #546 a cmake project is proposed, but no config is available with it

I haven't look at those projects, it may be so.

> What's your suggestion about this package?

As far as I understand there are a few forks on GitHub already. Please investigate what is the best for us and I will add new fork for the hunter-packages. If on your opinion they are not good for any reason I can create new repository `hunter-packages/rapidxml`.

ErniBrown commented:
Ok, my proposal is [timniederhausen/rapidxml](https://github.com/timniederhausen/rapidxml). It has a little bit more feature than required, test, documentation, but is one of the simplest one with cmake support.
I will add a config finder.
Can you please add a hunter-packages fork?

ruslo commented:
> Ok, my proposal is timniederhausen/rapidxml

Thanks, created: https://github.com/hunter-packages/rapidxml

I've added you to the team, you will be able to push commits after your accept an invitation (check your email). However it make sense to create pull requests for the review reasons. Also please read https://docs.hunter.sh/en/latest/creating-new/cmake-dependencies.html#maintenance

ErniBrown commented:
I switched to [hunter-packages/rapidxml](https://github.com/hunter-packages/rapidxml) and added a very simple RapidXMLConfig.cmake.
There's an issue: both release and debug configuration are "built", which is useless since it's a header only library: I was not able to correctly set **HUNTER_PACKAGE_CONFIGURATION_TYPES**, where should i change it?

ruslo commented:
> added a very simple RapidXMLConfig.cmake

It doesn't add an imported target. Just copy/paste the code I've mentioned earlier:
* https://github.com/ruslo/leathers/blob/1cf363e8abb440973cb400f421da8a3706a6b030/CMakeLists.txt#L91-L130

> both release and debug configuration are "built"

Since it's header only library I guess the performance is not an issue. We are just copying set of files twice.

> which is useless since it's a header only library

In this case probably yes, however in general even library is header only we may want to add debug macro for example, like RAPIDXML_DEBUG.

ErniBrown commented:
> It doesn't add an imported target. Just copy/paste the code I've mentioned earlier:

You are right, i fixed it.

ruslo commented on cmake/find/FindRapidXML.cmake:
```diff
@@ -0,0 +1,41 @@
+# ---------------------------------------------------------------------------
```
We don't need this file if RapidXMLConfig.cmake installed

ruslo commented on cmake/projects/RapidXML/hunter.cmake:
```diff
@@ -0,0 +1,19 @@
+)
+
+hunter_pick_scheme(DEFAULT url_sha1_cmake)
+hunter_download(PACKAGE_NAME RapidXML)
```
http://0.readthedocs.io/en/latest/git.html#check-there-is-a-new-line-at-end-of-file

ruslo commented on cmake/projects/RapidXML/hunter.cmake:
```diff
@@ -0,0 +1,19 @@
+    95164691ae21bd64b4dc0608a4e6d2d27c487dea
+)
+
+hunter_pick_scheme(DEFAULT url_sha1_cmake)
```
You should add `hunter_cacheable`, see https://docs.hunter.sh/en/latest/reference/user-modules/hunter_cacheable.html

ruslo commented on examples/RapidXML/CMakeLists.txt:
```diff
@@ -0,0 +1,19 @@
+# RapidXML available
+find_package(RapidXML REQUIRED CONFIG)
+
+add_executable(foo foo.cpp)
```
We should link `RapidXML::RapidXML` to the `foo` target.

ruslo replied:
The name of the target was my guess. I think the real name is `RapidXML::rapidxml`. To avoid such errors you can test changes with [jenkins.py](https://github.com/ruslo/hunter/wiki/dev.hunter.root#jenkinspy) script.

ErniBrown commented:
Everything fixed. Thanks for your patience, indeed I need a newline check script, i always forgot one somewhere

ErniBrown commented:
I decided to maintain RapidXML::RapidXML as target name, and fix it in hunter-packages/rapidxml, since there was a previous error in it.

ruslo commented:
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1188
* https://travis-ci.org/ingenue/hunter/builds/178850127

ruslo commented:
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1192
* https://travis-ci.org/ingenue/hunter/builds/178854011

ruslo commented:
Released: https://github.com/ruslo/hunter/releases/tag/v0.16.32

@ErniBrown feel free to create new wiki page
